frontend/src/components/DashboardProjectList.js

```diff
@@ -27,14 +38,22 @@ export default function DashboardProjectList(props) {
           <TableRow {...row.getRowProps()} onClick={() => {
             console.log(row.getRowProps());
             props.getproject(row)
-            }}>
+            }} className={classes.project}>
             {row.cells.map(cell => {
               return (
                 <TableCell {...cell.getCellProps()}>
                   {cell.render('Cell')}
                 </TableCell>
               )
             })}
+              {/* <InfoOutlinedIcon

```

Comment:
is there a reason that delete is inside the modal instead of to the side of the table as an extra column? would say that having the delete directly on the table makes more intuitive sense imo

Reply:
ah glad you asked :)
there's a few reasons I decided to do that
1. Your suggestion was actually my first idea, but I initially decided to scrap it because it made the table look a bit crowded. As it is, 1 button on every row is kinda pushing it. If I can figure out how to only make the buttons appear on the row that the user is currently hovering over, then (assuming I can resolve the other issues as well) I may consider reevaluating a separate delete button.
2. As I don't know exactly how pagination works yet and I haven't implemented event delegation yet for the buttons, I'm trying to avoid putting another event listener on every row of the table, as it slows down the page twice as fast.
3. Even though we automate a lot of the client's work, it still takes a lot of time to highlight everything, make classes, and create associations between the different highlights. Because the delete function is irreversible, I figured that burying the option (just slightly) would go a long way towards preventing very unfortunate mistakes. I would like to implement an undo feature (similar to what gmail does right after you send an email), but that's currently near the bottom of my todo list, so I chose to do this as a compromise in the meantime.
4. I'd argue that deletion could be considered a type of modification, which is what that modal popup is for. It's true however that the "i" icon typically means "show more information" as opposed to "modify". I'll have to look into whether a more intuitive alternative exists.

Reply:
Yea thats fine, just thought i'd ask. Ultimately, its up to Rob to decide if he wants any changes, although I imagine he should be fine with the current implementation. As per ur question regarding how you could do it, u could probably hardcode columns (TableCells) into the projectslist in each row and have empty string placeholders in the table header. Again tho, not necessary as I think the general functionality is there. Would say don't put too much emphasis on infinite scrolling -- while it would be nice to have it's not a feature expected by the client, try focusing on making sure all edge cases are considered and also that ur component is well documented.